### ISSUE TITLE
Testing speed on AWS GPU instance

### ISSUE THREAD

**Issue by vj-1988** [weiliu89/caffe#81]
I had deployed the pre-trained model (trained on VOC dataset, VGG16) on AWS GPU instance (g2.2xlarge ). The detection time is around 0.14-0.15 seconds with CUDNN v5. It is mentioned that SSD can perform 72 fps in TitanX. So is this speed on AWS instance normal or am I missing something?

I am training SSD (VGG16) on my custom dataset using the python ssd_pascal.py script. The training has crossed 20,000 iterations and the loss is around 0.8 and mbox_loss is also around 0.8. The loss started around 16.2 and mbox_loss at 16.4. I can see the training loss is gradually converging.  I have used around 150 images per class on average. Would it be sufficient to use around 150 images per class?

**Comment by weiliu89** [comment]
What is the GPU on AWS? It looks like it only has 1536 cores. Titan X has about 3584 cores. Besides, depend on the GPU architecture, the speed can be different as well. For example, Titan X is about 2x faster than K40. Besides, boost the clock speed can potentially increase the overall speed as well according to [here](http://caffe.berkeleyvision.org/performance_hardware.html).

**Comment by vj-1988** [comment]
The gpu is Grid k520. I expected around 15 fps minimum on Grid k520 (assuming TitanX is 4-5 times faster than k520) but I got around 6-7 fps.

**Comment by weiliu89** [comment]
I have provided a script to test the speed. You can run [this](https://github.com/weiliu89/caffe/blob/ssd/examples/ssd/ssd_pascal_speed.py).

That is hardware issue which is out of the scope of this repository. You should find reference to compare the speed between Titan X and other GPUs.

**Comment by vj-1988** [comment]
I am able to get around 64 fps in Titan X using python wrappers using batching.
